Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/overview/ContactDiaryOverviewFragment.kt

```diff
@@ -0,0 +1,88 @@
+package de.rki.coronawarnapp.contactdiary.ui.overview
+
+import android.os.Bundle
+import android.view.View
+import androidx.core.app.ShareCompat
+import androidx.fragment.app.Fragment
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.contactdiary.ui.overview.adapter.ContactDiaryOverviewAdapter
+import de.rki.coronawarnapp.databinding.ContactDiaryOverviewFragmentBinding
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.ui.doNavigate
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
+import timber.log.Timber
+import javax.inject.Inject
+
+class ContactDiaryOverviewFragment : Fragment(R.layout.contact_diary_overview_fragment), AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    @Inject lateinit var contactDiaryOverviewMenu: ContactDiaryOverviewMenu
+    private val vm: ContactDiaryOverviewViewModel by cwaViewModels { viewModelFactory }
+    private val binding: ContactDiaryOverviewFragmentBinding by viewBindingLazy()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val adapter = ContactDiaryOverviewAdapter {
+            vm.onItemPress(it)
+        }
+
+        setupToolbar()
+
+        binding.apply {
+            contactDiaryOverviewRecyclerview.adapter = adapter
+
+            contactDiaryOverviewHeader.contactDiaryHeaderButtonBack.buttonIcon.setOnClickListener {
+                vm.onBackButtonPress()
+            }
+        }
+
+        vm.listItems.observe2(this) {
+            adapter.setItems(it)
```

Comment:
Does this not need `notifyDataSetChanged`?

Reply:
yes.
its already handled inside of setItems(it)
_fun setItems(elements: List<ListItem>) {
        this.elements.clear()
        this.elements += elements
        notifyDataSetChanged()
    }_